Add Cyrillic keyboard layout support for flutter_tools terminal commands

777genius opened the pull request:
## Description

This PR adds support for Cyrillic (Russian ЙЦУКЕН) keyboard layout in the flutter_tools terminal handler, allowing users to use hot reload, hot restart, and other terminal commands without switching keyboard layouts.

## Motivation

Developers working with Cyrillic keyboard layouts currently must switch to a Latin layout every time they want to use terminal commands like hot reload (r/R) or help (h). This disrupts the development workflow and reduces productivity.

## Changes

The implementation maps Cyrillic characters to their corresponding Latin equivalents based on physical key positions on QWERTY/ЙЦУКЕН layouts:

- **к/К** → r/R (hot reload/restart)  
- **й/Й** → q/Q (exit)
- **ц/Ц** → w/W (dump widget tree)
- **в/В** → d/D (detach)
- **р/Р** → h/H (help)
- **ш/Ш** → i/I (widget inspector/invert images)
- **ы/Ы** → s/S (screenshot/semantics)
- **е/Е** → t/T (render tree)
- **м/М** → v/V (DevTools)
- **а** → f (focus tree)
- **п** → g (source generators)
- **Д** → L (layer tree)
- **щ/Щ** → o/O (platform toggle)
- **з/З** → p/P (debug paint/performance overlay)
- **Г** → U (semantics inverse order)
- **ф** → a (profile widgets)
- **и** → b (brightness)
- **с** → c (clear)

## Testing

- Added comprehensive test coverage for Cyrillic keyboard commands
- All existing terminal_handler tests pass
- Verified that Cyrillic characters trigger the same actions as their Latin equivalents

## Impact

This change improves the developer experience for users working with Cyrillic keyboard layouts by eliminating the need for constant layout switching during development. The implementation follows Flutter code style guidelines and does not affect existing functionality.

777genius commented on packages/flutter_tools/test/general.shard/terminal_handler_test.dart:
```diff
@@ -934,6 +934,93 @@ void main() {
+      await terminalHandler.processTerminalInput('Р');
+      expect(terminalHandler.residentRunner.hasHelpBeenPrinted, true);
+    });
   });
```
@gemini-code-assist fixed

777genius committed:
Add Cyrillic (Russian ЙЦУКЕН) keyboard layout support for terminal commands

Implements a keyboard layout mapping layer that allows users with non-Latin
layouts to use flutter_tools terminal commands without switching layouts.

The implementation uses a Map-based approach for character translation,
mapping characters based on physical key positions rather than linguistic
equivalents. This design makes it easy for contributors to add support for
additional layouts (Arabic, Hebrew, Greek, etc.).

Includes comprehensive test coverage following Flutter's testing patterns,
with tests for basic functionality, error handling (fatal/non-fatal), and
unsupported operations.

Fixes: #27021, #100456, #116658

bkonyi commented on packages/flutter_tools/lib/src/resident_runner.dart:
```diff
@@ -142,10 +142,11 @@ class FlutterDevice {
-            .file(globals.artifacts!.getHostArtifact(HostArtifact.flutterWebLibrariesJson))
-            .uri
-            .toString(),
+        librariesSpec:
```
I think you may have accidentally formatted these files with an older version of the Dart formatter. Can you upgrade your Flutter installation to the latest stable and format this again?

bkonyi reviewed (commented):
Just a few comments, but looks good overall!

777genius committed:
Address review comments for Cyrillic keyboard support

- Add warning comment about visually similar Cyrillic/Latin characters
- Remove redundant Cyrillic cases from switch (already handled by _mapKeyToLatin)
- Update docstring to reference _keyboardLayoutMappings
- Reformat with latest dart formatter

bkonyi committed:
Merge branch 'master' into feature/cyrillic-keyboard-support

777genius committed:
Add test to verify keyboard layout mapping keys are non-ASCII

- Format terminal_handler_test.dart
- Expose keyboardLayoutMappings with @visibleForTesting
- Add test to ensure mapping keys have code points > 127
  (prevents accidental addition of Latin characters as keys)
